src/Connection.php

```diff
@@ -159,7 +165,21 @@ private function receive($len = null)
     {
 
         if ($len) {
-            $line = fread($this->streamSocket, $len);
+            $chunkSize = $this->chunkSize;
+            $line = null;
+            $receivedBytes = 0;
+            while ($receivedBytes < $len) {
+                $bytesLeft = $len - $receivedBytes;
+                if ( $bytesLeft < 1500 ) {
+                    $chunkSize = $bytesLeft;
+                }
+
+                $line .= fread($this->streamSocket, $chunkSize);
+                $receivedBytes += $chunkSize;
+            }
+            if (strlen($line) > 2) {
+                $line = substr($line, 0, -2);
```

Comment:
Just had a look at the spec for MSG and the payload length doesn't include the \r\n at the end, from the example at http://nats.io/documentation/internals/nats-protocol/#a-name-msg-a-msg:4b9a176fc3df6b183e7b5ba0c60737ef.
I believe I added this line since initially after I had incorrectly set `if ( $bytesLeft < 1500 )` instead of the correct `if ( $bytesLeft < $this->chunksize )`, and that ended up reading in the \r\n at the end in cases.